### ISSUE TITLE
Add better descriptions as to what causes None to be returned. 

### ISSUE THREAD

**Issue by Peter-Lande** [dirs-dev/dirs-rs#44]
Currently the documentation is vague about what causes None returns in functions. It would be very helpful to be able to not have to comb through the code to implement the api calls and know what is happening if None is returned.

**Comment by opeolluwa** [comment]
Regards this, could you provide more information on the nature of what is to be returned

**Comment by opeolluwa** [comment]
Say, the format

**Comment by opeolluwa** [comment]
If you indulge me, I could return a struct, which will contain the PathBuf if any, and an error message if none

**Comment by soc** [comment]
Like a `Result`?

**Comment by opeolluwa** [comment]
A result that returns the Ok(Pathbuf) or an Err(Strong)

**Comment by Peter-Lande** [comment]
Yeah I think that is a better solution that the current Some/None strategy. Having some way to understand what failed to happen is better than it failing with no context. I am not entirely sure what the original context was of this issue, let me review the code I used that faced this issue and get back to you.

**Comment by opeolluwa** [comment]
I looked it up,  I think it's coming from the dir-sys crate

**Comment by soc** [comment]
The main concern when I thought about this pre-1.0 was that the errors that may show up in the failure case are hard to predict/test/standardize across platforms.

**Comment by opeolluwa** [comment]
Hmmm 🤔

**Comment by opeolluwa** [comment]
Now, the challenge is getting the error cause, which would be better than a generic message I suggested earlier
